lib/internal/modules/esm/loader.js

```diff
@@ -822,10 +842,13 @@ class ESMLoader {
 
     const nextResolve = nextHookFactory(chain, meta, validate);
 
-    const resolution = await nextResolve(originalSpecifier, context);
+    const resolution = nextResolve(originalSpecifier, context);
     const { hookErrIdentifier } = meta; // Retrieve the value after all settled
 
-    if (typeof resolution !== 'object') { // [2]
+    if (
+      typeof resolution !== 'object' || // [2]
+      resolution instanceof Promise // eslint-disable-line node-core/prefer-primordials
+    ) {
```

Comment:
```suggestion
    if (typeof resolution !== 'object') { // [2]
```

Reply:
This is necessary: it doesn't detect the promise without it.